datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/modular/AggregatePushDown.scala

```diff
@@ -106,12 +106,14 @@ trait AggregatePushDown { // self: ModularPlan =>
         } else {
           Map.empty[Int, (NamedExpression, Seq[NamedExpression])]
         }
-      case sum@AggregateExpression(Sum(Cast(expr, dataType, timeZoneId)), _, false, _)
-        if expr.isInstanceOf[Attribute] =>
-        val tAttr = selAliasMap.get(expr.asInstanceOf[Attribute]).getOrElse(expr)
+      case sum@AggregateExpression(Sum(child), _, false, _)
```

Comment:
Check the `CarbonExpressions.MatchCast` and use it here, no need to do if check

Reply:
This code use Cast with timeZoneId, But CarbonExpressions.MatchCast didn't have timeZoneId

Reply:
You are not using timeZoneId here, so you can use as below.
```
case sum@AggregateExpression(Sum(MatchCast(expr, dataType)), _, false, _) =>
        val tAttr = selAliasMap.get(expr.asInstanceOf[Attribute]).getOrElse(expr)
```

Reply:
No，origin line 114 ```val sum1 = AggregateExpression(Sum(Cast(tAttr, dataType, timeZoneId)), sum.mode, false)``` use it, if we don't pass the timeZoneId,  some test case will fail, such as "test create datamap with left join with group by with filter"

Reply:
Please try the following code
```
case sum@AggregateExpression(Sum(cast@MatchCast(expr, dataType)), _, false, _) =>
        val tAttr = selAliasMap.get(expr.asInstanceOf[Attribute]).getOrElse(expr)
          .asInstanceOf[Attribute]
        if (fact.outputSet.contains(tAttr)) {
          val sum1 = AggregateExpression(Sum(cast), sum.mode, false)
          val alias = Alias(sum1, sum1.toString)()
          val tSum = AggregateExpression(Sum(alias.toAttribute), sum.mode, false, sum.resultId)
          val (name, id) = aliasInfo.getOrElse(("", NamedExpression.newExprId))
          map += (ith -> (Alias(tSum, name)(exprId = id), Seq(alias)))
        } else {
          Map.empty[Int, (NamedExpression, Seq[NamedExpression])]
        }
```

Reply:
done